benchmark/README.md

```diff
@@ -49,79 +107,151 @@ The following build options are available:
 * `-DSWIFT_OPTIMIZATION_LEVELS`
     * A list of Swift optimization levels to build against
       (default: "O;Onone;Osize")
+* `-DSWIFT_DARWIN_XCRUN_TOOLCHAIN`
+    * The specific toolchain to use when looking up a clang compiler to use when
+      linking.
 * `-DSWIFT_BENCHMARK_EMIT_SIB`
     * A boolean value indicating whether .sib files should be generated
       alongside .o files (default: FALSE)
 
-The following build targets are available:
+The following build targets are available. They are not part of the all target,
+so one will need to pass them into ninja oneself.
 
 1. `swift-benchmark-macosx-x86_64`
 2. `swift-benchmark-iphoneos-arm64`
 3. `swift-benchmark-iphoneos-armv7`
 4. `swift-benchmark-appletvos-arm64`
 5. `swift-benchmark-watchos-armv7k`
 
-Build steps (with example options):
-
-1. `$ cd benchmark`
-2. `$ mkdir build`
-3. `$ cd build`
-4. `$ cmake ..`
-5. `$ make -j8 swift-benchmark-macosx-x86_64`
-
 Benchmark driver binaries are placed in `build/bin` and the required Swift
 standard library dylibs are placed in `build/lib`. The drivers dynamically link
 Swift standard library dylibs from a path relative to their location
 (../lib/swift) so the standard library should be distributed alongside them.
 
-Using the Benchmark Driver
---------------------------
+## Running Swift Benchmarks
+
+### Using build-script 
+
+To run Swift benchmarks, pass the `--benchmark` flag to `build-script`. The
+current benchmark results will be compared to the previous run's results if
+available. Results for each benchmark run are logged for future comparison.
+
+For branch based development, take a baseline benchmark on the Swift `master`
+branch, switch to a development branch containing potentially performance
+impacting changes, and run the benchmarks again. Upon benchmark completion, the
+benchmark results for the development branch will be compared to the most
+recent benchmark results for `master`.
+
+### Using standalone + swiftpm
+
+When building with SwiftPM, we only build a single binary of the low level
+driver (called SwiftBench) rather than compile one low level driver for each
+individual optimization setting. So it is important to make sure that if one is
+building in Xcode that one is building in release mode.
+
+In the case of building with swiftpm, the file will be in the directory .build
```

Comment:
How does one control the optimization level when building with `swiftpm`? Maybe an example here would help?

Reply:
Thats reasonable.